### PR TITLE
Configure iOS schemas to send data to MEP, remove legacy schemas

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T327341 and https://phabricator.wikimedia.org/T331481

### Notes
* This PR sets our app to send analytics data to the Modern Events Platform.
* It removes the base classes that served the legacy system, removes some funnels that were not to be moved into MEP, and updates the existing funnels to work with MEP
* I tried to merge most of the deletion commits, but the ones that were too far apart on the timeline were causing conflicts,  so there are some separate commits with deletions

### Test Steps
1. Run the app. The schemas for the explore feed, funnel, settings, login, readings lists, and user history (that captures a broader picture of the app's current state) were updated, so please interact with all these areas.
2. Check if info is being sent on a network debugger or the events platform.
3. A lot of code was removed, so a broader smoke test would be good while exploring the app

### CODE REVIEW

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 1]
```diff
@@ -100,36 +88,36 @@
...
-    @objc public func logUnsave(category: EventLoggingCategory, label: EventLoggingLabel?, articleURL: URL?) {
-        log(event(category: category, label: label, action: .unsave, measure: 1), language: articleURL?.wmf_languageCode)
+    public func logUnsave(category: EventCategoryMEP, label: EventLabelMEP?, articleURL: URL?) {
+        logEvent(action: .save, category: category, label: label, measure: 1)
     }
     
-    private func logUnsave(category: EventLoggingCategory, label: EventLoggingLabel? = nil, measure: Int = 1, language: String?, measureAge: Int? = nil, measurePosition: Int? = nil) {
-        log(event(category: category, label: label, action: .unsave, measure: measure, measureAge: measureAge, measurePosition: measurePosition), language: language)
+    private func logUnsave(category: EventCategoryMEP, label: EventLabelMEP? = nil, measure: Int = 1, language: String?, measureAge: Int? = nil, measurePosition: Int? = nil) {
+        logEvent(action: .save, category: category, label: label, measure: measure, measurePosition: measurePosition, measureAge: measureAge)
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:105]: Action should be .unsave

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 2]
```diff
@@ -100,36 +88,36 @@
...
     }
     
-    private func logUnsave(category: EventLoggingCategory, label: EventLoggingLabel? = nil, measure: Int = 1, articleURL: URL, measureAge: Int? = nil, measurePosition: Int? = nil) {
-        log(event(category: category, label: label, action: .unsave, measure: measure, measureAge: measureAge, measurePosition: measurePosition), language: articleURL.wmf_languageCode)
+    private func logUnsave(category: EventCategoryMEP, label: EventLabelMEP? = nil, measure: Int = 1, articleURL: URL, measureAge: Int? = nil, measurePosition: Int? = nil) {
+        logEvent(action: .save, category: category, label: label, measure: measure, measurePosition: measurePosition, measureAge: measureAge)
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:109]: Action should be .unsave

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 3]
```diff
@@ -100,36 +88,36 @@
...
     }
 
-    public func logUnsave(category: EventLoggingCategory, label: EventLoggingLabel? = nil, measure: Int = 1, articleURL: URL, date: Date?, measurePosition: Int) {
-        log(event(category: category, label: label, action: .unsave, measure: measure, measureAge: daysSince(date), measurePosition: measurePosition), language: articleURL.wmf_languageCode)
+    public func logUnsave(category: EventCategoryMEP, label: EventLabelMEP? = nil, measure: Int = 1, articleURL: URL, date: Date?, measurePosition: Int) {
+        logEvent(action: .save, category: category, label: label, measure: measure, measurePosition: measurePosition, measureAge: daysSince(date))
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:113]: Action should be .unsave

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 4]
```diff
@@ -139,48 +127,49 @@
...
     public func logCreateInReadingLists() {
-        log(event(category: .saved, label: .lists, action: .createList, measure: 1))
+        logEvent(action: .createList, category: .saved, label: .lists, measure: 1)
     }
     
     // - MARK: Add articles to reading list
     
     public func logDeleteInAddToReadingList(readingListsCount: Int = 1) {
-        log(event(category: .addToList, label: nil, action: .deleteList, measure: readingListsCount))
+        logEvent(action: .deleteList, category: .addToList, label: .lists, measure: readingListsCount)
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:143]: Label was nil before, not sure if that was right though.

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 5]
```diff
@@ -139,48 +127,49 @@
...
     }
     
     public func logCreateInAddToReadingList() {
-        log(event(category: .addToList, label: nil, action: .createList, measure: 1))
+        logEvent(action: .createList, category: .addToList, label: .lists, measure: 1)
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:147]: Label was nil before.

File: Wikipedia/Code/ReadingListsFunnel.swift [hunk 6]
```diff
@@ -139,48 +127,49 @@
...
     // - MARK: Import Shared Reading Lists
     
     public func logStartImport(articlesCount: Int) {
-        log(event(category: .shared, label: nil, action: .receiveStart, measure: articlesCount))
+        logEvent(action: .receiveStart, category: .shared, label: nil, measure: articlesCount)
     }
     
     public func logCancelImport() {
-        log(event(category: .shared, label: nil, action: .receiveCancel))
+        logEvent(action: .receiveCancel, category: .shared, label: nil, measure: 1)
```
Comment (tonisevener) [inline Wikipedia/Code/ReadingListsFunnel.swift:157]: Didn't pass in measure before.

**Review by tonisevener — COMMENTED** [review]
Thanks for working through all of this! 😅

I've only reviewed the first two commits, but I thought I'd post the things that jumped out at me so far. I'll continue reviewing it tomorrow.

File: Wikipedia/Code/SearchFunnel.swift [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    func logSearchDidYouMean(source: String) {
+        logEvent(action: .didYouMean, source: source)
+    }
+
+    func logSearchResultTap(position: Int, source: String) {
+        logEvent(action: .click, source: source, position: position)
+    }
+
+    func logSearchCancel(source: String) {
+        logEvent(action: .click, source: source)
```
Comment (tonisevener) [inline Wikipedia/Code/SearchFunnel.swift:59]: Need .cancel action.

File: WMF Framework/EventLogging.xcdatamodeld/.xccurrentversion [hunk 8]
```diff
@@ -1,8 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (tonisevener) [inline WMF Framework/EventLogging.xcdatamodeld/.xccurrentversion:1]: Can you remove this file entirely? It's not in the project anymore but I still see it in the WMF Framework directory.

It also may be worthwhile to add a migration step in `performUpdatesFromLibraryVersion` to delete the `sqlite` files in the app container.

File: Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocController.swift [hunk 9]
```diff
@@ -68,27 +68,7 @@ class ArticleAsLivingDocController: NSObject {
     
     var articleAsLivingDocViewController: ArticleAsLivingDocViewController?
     
-    var shouldAttemptToShowArticleAsLivingDoc: Bool {
```
Comment (tonisevener) [inline Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocController.swift:71]: Solving in this way should be fine. It looks like the old `EventLoggingService` was used as storage for which A/B testing bucket a user landed in for the article as a living document experiment.

**Review by tonisevener — CHANGES_REQUESTED** [review]
This wraps up the initial review!

File: Wikipedia/Code/SessionsFunnel.swift [hunk 10]
```diff
@@ -9,67 +7,49 @@
...
+            label: label, measure_time: Int(round(measure ?? Double())), page_load_latency_min: Int(round(pageLoadMin ?? Double())), page_load_latency_max: Int(round(pageLoadMax ?? Double())), page_load_latency_average: Int(round(pageLoadAverage ?? Double())) )
+
+        EventPlatformClient.shared.submit(stream: .sessions, event: event1)
     }
-    
+
     @objc public func logSessionStart() {
         resetSession()
-        log(event(category: .unknown, label: nil, action: .sessionStart))
+        logEvent(category: .unknown, label: nil, action: .sessionStart)
```
[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/SessionsFunnel.swift:36]: This comment may be outdated after you do the work mentioned in https://phabricator.wikimedia.org/T327341#8674562.

File: Wikipedia/Code/SessionsFunnel.swift [hunk 11]
```diff
@@ -9,67 +7,59 @@
...
     }
-    
-    @objc public func logSessionStart() {
-        resetSession()
-        log(event(category: .unknown, label: nil, action: .sessionStart))
+
+    @objc public func setSessionStart() {
+        if EventPlatformClient.shared.hasSessionTimedOut {
+            defer {
+                resetSession()
```
Comment (tonisevener) [inline Wikipedia/Code/SessionsFunnel.swift:37]: SessionsFunnel being able to reset EventPlatform's session logic still feels a bit iffy to me and could be the source of future bugs. I'm struggling with how best to clean it up though. Maybe once that is the last problem to solve, we can tackle it in a followup PR.

File: Wikipedia/Code/ArticleLocationCollectionViewController.swift [hunk 12]
```diff
@@ -79,32 +74,22 @@ class ArticleLocationCollectionViewController: ColumnarCollectionViewController,
...
     override func shareArticlePreviewActionSelected(with articleController: ArticleViewController, shareActivityController: UIActivityViewController) {
-        guard let context = feedFunnelContext else {
-            super.shareArticlePreviewActionSelected(with: articleController, shareActivityController: shareActivityController)
-            return
-        }
         super.shareArticlePreviewActionSelected(with: articleController, shareActivityController: shareActivityController)
-        FeedFunnel.shared.logFeedDetailShareTapped(for: context, index: previewedIndexPath?.item, midnightUTCDate: context.midnightUTCDate)
     }
 
     override func readMoreArticlePreviewActionSelected(with articleController: ArticleViewController) {
```
Comment (tonisevener) [inline Wikipedia/Code/ArticleLocationCollectionViewController.swift:80]: No need to override at all anymore if you're only calling `super`, you can remove this method.

File: Wikipedia/Code/SessionsFunnel.swift (hunk 11 above)
Comment (mazevedofs) [reply Wikipedia/Code/SessionsFunnel.swift:37]: I've separated the sessionID creation logic into another class, but I'm still struggling to find a better way to handle this bit without creating a tight coupling of the new `UserSession` class and the `SessionsFunnel`

**Comment by mazevedofs** [comment]
Hi @tonisevener! I think it's ready for another review. I'm still not 100% happy with the session handling, so let me know if you can think of a better approach.

**Pull request merged**